specs/AppLifecycle/Instancing/AppLifecycle SingleMulti-Instancing.md

```diff
@@ -0,0 +1,422 @@
+normal app initialization. In this example, the app signals that an instance is reusable by
+specifying the "reusable" text as the key it registers with the system. Of course, exactly what text
+to use for this purpose is entirely up to the app to decide. For more complex scenarios, if the app
+wanted to allow for multiple reusable instances, it could specify "reusable" plus some unique
+identifier for each instance, and so on.
+
+```c++
+int APIENTRY B_wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
+    ActivationArguments activationArgs =
+        AppInstance::GetCurrent().GetActivatedEventArgs();
+
+    // As above, check for any specific activation kind we care about.
+    ExtendedActivationKind kind = activationArgs.Kind;
+    if (kind == ExtendedActivationKind::File)
+    {
+        // etc... as in previous scenario.
+    }
+    else
+    {
+        // For other activation kinds, we'll trawl all instances to see if
+        // any are suitable for redirecting this request. First, get a list
+        // of all running instances of this app.
+        auto instances = AppInstance::GetInstances();
+
+        // In the simple case, we'll redirect to any other instance.
+        AppInstance instance = instances.GetAt(0);
+
+        // If the app re-registers re-usable instances, we can filter for these instead.
+        bool isFound = false;
+        for (AppInstance instance : instances)
+        {
+            if (instance.Key.c_str() == "REUSABLE")
+            {
+                isFound = true;
+                instance.RedirectTo(activationArgs);
+                break;
+            }
+        }
+        if (!isFound)
+        {
+            // We'll register this as a reusable instance, and then
+            // go ahead and do normal initialization.
+            winrt::hstring szKey = L"REUSABLE";
+            AppInstance::FindOrRegisterForKey(szKey);
+            RegisterClassAndStartMessagePump(hInstance, nCmdShow);
+        }
+    }
+    return 1;
+}
+```
+
+### Handling redirection
+
+The first 2 scenarios illustrate how an app would redirect to another running instance. This
+scenario shows how an app would handle the case where it is already running, and then a fresh
+activation is redirected to this instance. The AppInstance class will expose an Activated event, for
+which the app can register. In its Activated event handler, the app would examine the type of
+activation, retrieve the rich ActivationArguments object, and then do whatever app-specific work it
+wishes to do based on those arguments.
+
+```c++
+int APIENTRY wWinMain(
+    _In_ HINSTANCE hInstance, _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR lpCmdLine, _In_ int nCmdShow)
+{
+    // First, hook up the Activated event, to allow for this instance of the app
+    // getting reactivated as a result of multi-instance redirection.
+    AppInstance::Activated([](ActivationArguments args)
```

Comment:
We seem to be inconsistent about which things are members of the singleton AppInstance object and which are statics. Should pick a pattern and apply consistently.

Reply:
RECOMMEND: Typo from the prior version of the spec - the Activated event goes on an instance, which you get from `::GetCurrent()`

Reply:
Fixed.